wgsl/index.bs

```diff
@@ -3771,6 +3772,14 @@ Issue: Which index is used when it's out of bounds?
            (OpAccessChain)
 </table>
 
+Note: Reflecting the limitations of the languages WGSL is meant to be translated
+into, it is only possible to use dynamically computed indices to subscript
+references to matrices. A matrix not behind a reference may only be indexed by a
+constant. To work around this restriction, consider storing the matrix in a
```

Comment:
Let's be careful with "constant" part. An overridable constant is not a literal/constexpr